base/io.jl

```diff
@@ -167,24 +167,54 @@ The text is assumed to be encoded in UTF-8.
 readuntil(filename::AbstractString, args...) = open(io->readuntil(io, args...), filename)
 
 """
-    readline(stream::IO=STDIN)
-    readline(filename::AbstractString)
+    readline(stream::IO=STDIN, chomp::Bool=true)
+    readline(filename::AbstractString, chomp::Bool=true)
 
-Read a single line of text, including a trailing newline character (if one is reached before
-the end of the input), from the given I/O stream or file (defaults to `STDIN`).
-When reading from a file, the text is assumed to be encoded in UTF-8.
+Read a single line of text from the given I/O stream or file (defaults to `STDIN`).
+Lines in the input can end in `'\\n'` or `"\\r\\n"`. When reading from a file, the text is
+assumed to be encoded in UTF-8.
+
+If `chomp=false` trailing newline character(s) will be included in the output
+(if reached before the end of the input); otherwise newline characters(s)
+are stripped from result.
 """
-readline(filename::AbstractString) = open(readline, filename)
+function readline(filename::AbstractString, chomp::Bool=true)
+    open(filename) do f
+          readline(f, chomp)
+    end
+end
+readline() = readline(STDIN, false)
+
+function readline(s::IO, chomp::Bool=true)
+    chomp || return String(readuntil(s, 0x0a))
```

Comment:
I rather find it harder to read since all steps from the `chomp=false` case are repeated. Other than that, the PR looks good to me.